server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/FilterTest.java

```diff
@@ -572,4 +591,1159 @@ public void allFieldsAndComparatorsShouldBeSupported() {
             .body(ARGUMENTS + ".singleton[4].condition.comparator", equalTo("contains"));
     }
 
+    @Test
+    public void messageShouldBeAppendedInSpecificMailboxWhenFromRuleMatches() {
+        with()
+            .header("Authorization", accessToken.serialize())
+            .body("[[" +
+                "  \"setFilter\", " +
+                "  {" +
+                "    \"singleton\": [" +
+                "    {" +
+                "      \"id\": \"3000-345\"," +
+                "      \"name\": \"Emails from bob\"," +
+                "      \"condition\": {" +
+                "        \"field\": \"from\"," +
+                "        \"comparator\": \"contains\"," +
+                "        \"value\": \"" + BOB + "\"" +
+                "      }," +
+                "      \"action\": {" +
+                "        \"appendIn\": {" +
+                "          \"mailboxIds\": [\"" + matchedMailbox.serialize() + "\"]" +
+                "        }" +
+                "      }" +
+                "    }" +
+                "  ]}, " +
+                "\"#0\"" +
+                "]]")
+            .post("/jmap");
```

Comment:
You may assert at least the status code of this response.
If this request fails then it will be hard to understand why the assert of this test is failing.